h2o-automl/src/main/java/ai/h2o/automl/targetencoding/TargetEncoderModel.java

```diff
@@ -0,0 +1,158 @@
+package ai.h2o.automl.targetencoding;
+
+import hex.Model;
+import hex.ModelCategory;
+import hex.ModelMetrics;
+import hex.ModelMojoWriter;
+import water.H2O;
+import water.Iced;
+import water.Key;
+import water.MRTask;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.util.IcedHashMap;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class TargetEncoderModel extends Model<TargetEncoderModel, TargetEncoderModel.TargetEncoderParameters, TargetEncoderModel.TargetEncoderOutput> {
+
+  public TargetEncoderModel(Key<TargetEncoderModel> selfKey, TargetEncoderParameters parms, TargetEncoderOutput output) {
+    super(selfKey, parms, output);
+  }
+
+  @Override
+  public ModelMetrics.MetricBuilder makeMetricBuilder(String[] domain) {
+    throw H2O.unimpl("No Model Metrics for TargetEncoder.");
+  }
+
+  @Override
+  protected double[] score0(double[] data, double[] preds) {
+    return new double[0];
+  }
+
+  private static IcedHashMap<String, Map<String, TEComponents>> _targetEncodingMap;
+
+
+  public static class TargetEncoderParameters extends Model.Parameters {
```

Comment:
Any reason why it's `static`?

Reply:
IMHO this is the right way... the automatic mapping from Schema objects to Parameter objects wouldn't work without it. Plus you don't want to serialize the outer model if you just want to send the parameters to workers.

Reply:
@michalkurka In the beginning  I was actually trying to avoid making it static but failed because of the reasons you have just described. Then I got it :)